svm/src/transaction_processor.rs

```diff
@@ -208,7 +206,6 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
 
     /// Returns the current environments depending on the given epoch
     /// Returns None if the call could result in a deadlock
-    #[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
```

Comment:
instead of removing `qualifier_attr`, can we find a way to remove the default `pub` visibility of `get_environments_for_epoch()`?

Reply:
it looks like it needs to be `pub`. It's used in the `pub` method `Bank::load_program` which is in a different crate https://github.com/anza-xyz/agave/blob/bd1a5dfba3886f4ea95d1c6226785fae3ccbbc40/runtime/src/bank.rs#L7036

Reply:
Will it help if `Bank::load_program()` were to be put under the same constraint for its visibility?

Reply:
load_program() is used by ledger-tool so it too needs to be pub

So this is code that doesn't belong under dev-context-only-utils anymore; if it is to be put behind a feature gate then it would need to be a new feature

Reply:
Changing the load_program() to the following still allows ledger-tool to compile.
```
    #[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
    fn load_program(
```

But, removing `pub` from `get_environments_for_epoch()` doesn't let it stay visible to `load_program()`. 
Note that `load_program()` is already under 
```
#[cfg(feature = "dev-context-only-utils")]
impl Bank {
```

The reason to add the conditional visibility (even though it is broken right now) to `get_environments_for_epoch()` was due to a potential footgun in the function. So if possible, we should try to make this conditional visibility work, instead of removing it.

Reply:
> Note that load_program() is already under [...]
Oh right I missed this